### PR TITLE
Receive files outside the classpath in quarkus-liquibase

### PR DESCRIPTION
**Pull request by juanjgarciavilla**
This PR is the continuation of an improvement made in quarkus-liquibase-mongodb extension which was replicated in quarkus-liquibase extension, more information about it can be found in this [issue](https://github.com/quarkusio/quarkus/issues/39683).

### CODE REVIEW

File: extensions/liquibase-mongodb/runtime/src/main/java/io/quarkus/liquibase/mongodb/LiquibaseMongodbFactory.java [hunk 1]
```diff
@@ -51,6 +51,8 @@ private ResourceAccessor resolveResourceAccessor() throws FileNotFoundException
         }
 
         CompositeResourceAccessor compositeResourceAccessor = new CompositeResourceAccessor();
+        compositeResourceAccessor
+                .addResourceAccessor(new ClassLoaderResourceAccessor(Thread.currentThread().getContextClassLoader()));
```
Comment (geoand) [inline extensions/liquibase-mongodb/runtime/src/main/java/io/quarkus/liquibase/mongodb/LiquibaseMongodbFactory.java:55]: Why this change?

Comment (juanjogv) [reply extensions/liquibase-mongodb/runtime/src/main/java/io/quarkus/liquibase/mongodb/LiquibaseMongodbFactory.java:55]: It was something I realized along the way that it can be useful to have a changelog within the classpath that references files on and off the classpath as shown below.

image

Comment (geoand) [reply extensions/liquibase-mongodb/runtime/src/main/java/io/quarkus/liquibase/mongodb/LiquibaseMongodbFactory.java:55]: Thanks for the information.

Comment (geoand) [reply extensions/liquibase-mongodb/runtime/src/main/java/io/quarkus/liquibase/mongodb/LiquibaseMongodbFactory.java:55]: In this case I think this change should be a separate commit as it's pretty much unrelated to the rest of the changes in PR

Comment (juanjogv) [reply extensions/liquibase-mongodb/runtime/src/main/java/io/quarkus/liquibase/mongodb/LiquibaseMongodbFactory.java:55]: Understood, but then I put it in a different commit or in a different pr?

**Review by geoand — APPROVED** [review]
Thanks!

File: extensions/liquibase/runtime/src/main/java/io/quarkus/liquibase/runtime/LiquibaseDataSourceBuildTimeConfig.java [hunk 2]
```diff
@@ -28,4 +32,9 @@ public static final LiquibaseDataSourceBuildTimeConfig defaultConfig() {
     @ConfigItem(defaultValue = DEFAULT_CHANGE_LOG)
     public String changeLog;
 
+    /**
+     * The search path for DirectoryResourceAccessor
+     */
+    @ConfigItem(defaultValue = "/")
```
[the remark this answers is not in the document]
Comment (geoand) [reply extensions/liquibase/runtime/src/main/java/io/quarkus/liquibase/runtime/LiquibaseDataSourceBuildTimeConfig.java:38]: Yeah I saw that but I have to admit it's confusing. Anyway, let's leave it like this for now, but I assume that using `Optional<List<String>>` for `searchPath` would be better.

[the remark this answers is not in the document]
Comment (juanjogv) [reply extensions/liquibase/runtime/src/main/java/io/quarkus/liquibase/runtime/LiquibaseDataSourceBuildTimeConfig.java:38]: I did it this way because at the time I could not find how to declare the default list as an empty list in the build time properties, because if I put a default value of "" in the tag, that is an empty string, I got an error related to null.

[the remark this answers is not in the document]
Comment (geoand) [reply extensions/liquibase/runtime/src/main/java/io/quarkus/liquibase/runtime/LiquibaseDataSourceBuildTimeConfig.java:38]:
```java
    /**
     * The search path for DirectoryResourceAccessor
     */
    @ConfigItem
    public Optional<List<String>> searchPath;
```

should work, no?

[the remark this answers is not in the document]
Comment (geoand) [reply extensions/liquibase/runtime/src/main/java/io/quarkus/liquibase/runtime/LiquibaseDataSourceBuildTimeConfig.java:38]: Let's merge this and you can try and update both configs with my suggestion. If all goes well, you can open a new PR
